src/extension.ts

```diff
@@ -187,6 +199,17 @@ function initializeCredentialsProviderManager() {
     CredentialsProviderManager.getInstance().addProviderFactory(new SharedCredentialsProviderFactory())
 }
 
+function makeEndpointsProvider(): EndpointsProvider {
+    const localManifestFetcher = new FileResourceFetcher(ext.manifestPaths.endpoints)
+    const remoteManifestFetcher = new HttpResourceFetcher(endpointsFileUrl)
+
+    const provider = new EndpointsProvider(localManifestFetcher, remoteManifestFetcher)
+    // tslint:disable-next-line:no-floating-promises -- start the load without waiting. It raises events as fetchers retrieve data.
+    provider.load()
```

Comment:
Interesting, I definitely have some questions around this:
* What's the worst-case scenario here? Is there a chance that an endpoint isn't loaded by the time the user wants to use this? 
* What's the performance gain over an await?
* Is this a pattern we want to use going forward?

Reply:
On systems with slow internet connections, you do not want to degrade the toolkit activation experience. This is why we do not await here. This offers a performance gain against the existing experience.

Endpoints data is used to boot critical parts of the toolkit, like the AWS Explorer. That is why it was changed to load local sources asap, then remote resources with more tolerance. The EndpointsProvider event is the mechanism for components to receive the updated endpoints if/when/as they arrive.

Reply:
What's the experience if a user has a 10000+ ping? Do we have to signal that this is done loading? What if there's a failure; do we have a retry strategy?

Reply:
The current approach is that the bundled file is a reasonable fallback. The toolkits today don't do anything more complex like retry.

Signalling is done at the discretion of any system registering to the EndpointsProvider event. Today I feel there is no component that needs to signal the user.